catalog/cataloger_merge.go

```diff
@@ -52,49 +54,46 @@ func (c *cataloger) Merge(ctx context.Context, repository, leftBranch, rightBran
 				Limit: -1,
 			},
 		}
-
-		scanner, err := NewDiffScanner(tx, params)
+		relation, err := getRefsRelationType(tx, params)
```

Comment:
guess this code need to get the update from the parent branch first - we can get the refs relation from the scanner